platform-hub-api/app/jobs/privileged_token_expirer_job.rb

```diff
@@ -0,0 +1,50 @@
+class PrivilegedTokenExpirerJob < ApplicationJob
+  queue_as :tokens_expire
+
+  IDENTITY_BATCH_SIZE = 100
+
+  def self.is_already_queued
+    Delayed::Job.where(queue: :tokens_expire).count > 0
+  end
+
+  def perform
+    return unless FeatureFlagService.is_enabled?(:kubernetes_tokens)
+    
+    Identity.kubernetes.find_each(batch_size: IDENTITY_BATCH_SIZE) do |i|
+      should_update = false
+      deescalated_token_group_clusters = []
+
+      tokens = Kubernetes::TokenService.tokens_from_identity_data(i.data).each do |token|
+        next if token.expire_privileged_at.nil? || Time.parse(token.expire_privileged_at).future?
+
+        # Remove ALL privileged groups from user token groups and reset expiration timestamp
+        token.groups = token.groups - Kubernetes::TokenGroupService.privileged_group_ids
```

Comment:
Could cache the value of `Kubernetes::TokenGroupService.privileged_group_ids` at the beginning?